subprojects/docs/src/docs/userguide/kotlin_dsl.adoc

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[kotlin_dsl]]
+= Gradle Kotlin DSL
+
+
+[TIP]
+====
+If you are looking into migrating an existing build to the Gradle Kotlin DSL, please check out the link:https://guides.gradle.org/migrating-build-logic-from-groovy-to-kotlin/[migration guide].
+====
+
+
+[[sec:kotlin-dsl_plugin]]
+== The `kotlin-dsl` plugin
+
+When applied to a project, the `kotlin-dsl` Gradle plugin configures the project's Kotlin source sets to enable Gradle Kotlin DSL features.
+
+It is useful for all projects implementing build logic: <<organizing_gradle_projects.adoc#sec:build_sources, buildSrc>>, <<composite_builds.adoc, included builds>> for build logic or link:{guidesUrl}/designing-gradle-plugins/[Gradle plugin] builds.
```

Comment:
:x: The link `<<composite_builds.adoc` needs to be `<<composite_builds.adoc#composite_builds` in order to work in all envs.

Reply:
Good to know, thanks! And fixed